api/util.js

```diff
@@ -48,7 +48,11 @@ export function isAsyncFunction (object) {
 }
 
 export function isArgumentsObject (object) {
-  return isPlainObject(object)
+  return (
+    !Array.isArray(object) &&
+    isPlainObject(object) &&
+    Number.isFinite(object.length)
```

Comment:
Just curious, why is las check necessary? Also it could be more strict probably (`Number.isInteger`)